modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteNodeRestartTest.java

```diff
@@ -1230,13 +1226,11 @@ private void createTableWithData(List<IgniteImpl> nodes, String name, int replic
         }
     }
 
-    private void waitForIndex(Collection<IgniteImpl> nodes, String indexName) throws InterruptedException {
-        // FIXME: Wait for the index to be created on all nodes,
-        //  this is a workaround for https://issues.apache.org/jira/browse/IGNITE-18733 to avoid missed updates to the PK index.
+    private void waitForIndexToBecomeAvailable(Collection<IgniteImpl> nodes, String indexName) throws InterruptedException {
         assertTrue(waitForCondition(
                 () -> nodes.stream()
                         .map(nodeImpl -> nodeImpl.catalogManager().index(indexName.toUpperCase(), nodeImpl.clock().nowLong()))
-                        .allMatch(Objects::nonNull),
+                        .allMatch(indexDescriptor -> indexDescriptor != null && !indexDescriptor.writeOnly()),
```

Comment:
I think it's better to use `org.apache.ignite.internal.catalog.events.CatalogEvent#INDEX_AVAILABLE`.

Reply:
The event only brings index ID, not its name, but we only have name here, and to find index ID we must wait till the index descriptor appears in the catalog; so using the event would just complicate things without any benefits.